lib/src/state_manager/simple/get_state.dart

```diff
@@ -8,39 +8,109 @@ import 'package:get/state_manager.dart';
+
+/// Complies with [GetStateUpdater]
+///
+/// This mixin's function represents a [GetStateUpdater], and might be used
+/// by [GetBuilder()], [SimpleBuilder()] (or similar) to comply
+/// with [GetStateUpdate] signature. REPLACING the [StateSetter].
+/// Avoids the potential (but extremely unlikely) issue of having
+/// the Widget in a dispose() state, and abstracts the API from the ugly fn((){}).
+/// TODO: check performance HIT for the extra method call.
+///
+mixin GetStateUpdaterMixin<T extends StatefulWidget> on State<T> {
+  bool getUpdate() {
+    final _mounted = mounted;
+    if (_mounted) setState(() {});
+    return _mounted;
+  }
+}
 
 class GetxController extends DisposableInterface {
-  final HashSet<StateSetter> _updaters = HashSet<StateSetter>();
+  final _updaters = HashSet<GetStateUpdate>();
+
+//  final _updatersIds = HashMap<String, StateSetter>(); //<old>
+  final _updatersIds = HashMap<String, GetStateUpdate>();
+
+//  final _updatersGroupIds = HashMap<String, List>(); //<old>
+  final _updatersGroupIds = HashMap<String, HashMap<int, GetStateUpdate>>();
 
-  final HashMap<String, StateSetter> _updatersIds =
-      HashMap<String, StateSetter>();
+  static int _groupIdCount = 0;
 
-  /// Update GetBuilder with update();
+  /// Rebuilds [GetBuilder] each time you call [update()];
+  /// Can take a List of [ids], that will only update the matching
+  /// `GetBuilder( id: )`,
+  /// [ids] can be reused among `GetBuilders` like group tags.
+  /// The update will only notify the Widgets, if [condition] is true.
   void update([List<String> ids, bool condition = true]) {
     if (!condition) return;
-    (ids == null)
-        ? _updaters.forEach((rs) => rs(() {}))
-        : ids.forEach((element) {
-            _updatersIds[element]?.call(() {});
-          });
+    if (ids == null) {
+//      _updaters.forEach((rs) => rs(() {}));//<old>
+      _updaters.forEach((rs) => rs());
+    } else {
+      ids.forEach((element) {
+//        _updatersGroupIds[element]?.forEach((k, rs) => rs(() {}));//<old>
+//        _updatersIds[element]?.call(() {});//<old>
+        _updatersIds[element]?.call();
+        _updatersGroupIds[element]?.forEach((k, rs) => rs());
+      });
+    }
   }
 
-  Disposer addListener(StateSetter listener) {
+//  VoidCallback addListener(StateSetter listener) {//<old>
+  VoidCallback addListener(GetStateUpdate listener) {
     _updaters.add(listener);
     return () => _updaters.remove(listener);
   }
 
-  // void removeListener(StateSetter listener) {
-  //   _updaters.remove(listener);
-  // }
+//  VoidCallback addListenerId(String key, StateSetter listener) {//<old>
+  VoidCallback addListenerId(String key, GetStateUpdate listener) {
+//    _printCurrentIds();
+    if (_updatersIds.containsKey(key)) {
+      final _innerKey = _groupIdCount++;
+//      final _ref = _updatersGroupIds[key] ??= HashMap<int, StateSetter>();//<old>
+      final _ref = _updatersGroupIds[key] ??= HashMap<int, GetStateUpdate>();
```

Comment:
In my tests the update with ID is slower than before.
Just about the Iterables speed issue:
HashSet is the fastest of all in any situation.
LinkedList is fast for a few items, and HashMap starts to take advantage from the tenth item.
I believe that a HashMap<String, HashSet> could reduce that difference dramatically, at least in theory.

Reply:
@jonataslaw . Ok, if that's the only change, i will push it soon.